src/components/data/RenameTableModal.js

```diff
@@ -0,0 +1,61 @@
+import { Fragment, useState } from 'react'
+import { h } from 'react-hyperscript-helpers'
+import { ButtonPrimary, IdContainer, spinnerOverlay } from 'src/components/common'
+import { ValidatedInput } from 'src/components/input'
+import Modal from 'src/components/Modal'
+import { Ajax } from 'src/libs/ajax'
+import { reportError } from 'src/libs/error'
+import Events from 'src/libs/events'
+import { FormLabel } from 'src/libs/forms'
+import { useCancellation } from 'src/libs/react-utils'
+
+
+export const tableNameInput = ({ inputProps, ...props }) => h(ValidatedInput, {
+  ...props,
+  inputProps: {
+    ...inputProps,
+    autoFocus: true,
+    placeholder: 'Enter a name'
+  }
+})
+
+const RenameTableModal = ({ onDismiss, onSuccess, namespace, name, selectedDataType }) => {
+  // State
+  const [newName, setNewName] = useState('')
+  const [renaming, setRenaming] = useState(false)
+
+  const signal = useCancellation()
+
+  return h(Modal, {
+    onDismiss,
+    title: 'Rename Data Table',
+    okButton: h(ButtonPrimary, {
+      disabled: renaming,
+      onClick: async () => {
+        setRenaming(true)
```

Comment:
You should likely wrap this in withErrorReporting instead of doing the try/catch (and that makes sure that the Metrics call is also wrapped).

Reply:
Thank you, that's much cleaner. Addressed in 4b3ab4bace2b9c27e8212700ab395fb3136f3cb3